### DISCUSSION TITLE
Set up problem

### DISCUSSION THREAD

**Discussion by Rynf0rce** [jeremylvln/Shulker#637] [Q&A] [answered]
Hi, sorry for the question right off the bat

I installed shulker as Helm after installing Agones in EKS, but faced the following error in the pod. (https://shulker.jeremylvln.fr/latest/guide/getting-started/installation.html)

Maybe it's because I don't know K8S very well, but I don't know how to solve this problem.

I have finshed set secrets, allocator-client.default (https://agones.dev/site/docs/advanced/allocator-service/#client-certificate)

<img width="616" alt="Screenshot 2024-08-27 at 14 45 00" src="https://github.com/user-attachments/assets/f642c67b-3ffe-43e6-9286-e6fcf151d2f0">

kubectl describe pod -n shulker-system shulker-shulker-operator-7c5cfcb4cd-ncm46'

  Warning  FailedMount  34s (x15 over 14m)  kubelet            MountVolume.SetUp failed for volume "agones-allocator-client-tls" : secret "allocator-client.default" not found

**Comment by jeremylvln** [comment, marked as answer]
Hi @risograph !

Are you sure you configured Agones correctly? Shulker is looking for this secret in the namespace it's deployed.

So you need to make sure this same namespace is listed in the Agones's Helm configuration at `gameservers.namespaces`.

If you deployed Shulker in, let's say `shulker-system`, your Agones Helm `values.yaml` file should contain something like:

```yaml
gameservers:
  namespaces:
    - shulker-system
```

As long as you don't see the secret in Shulker's namespace, it means that you have an issue with Agones (if you have still nothing, look at their logs, maybe it will contain a clue...)

**Reply by Rynf0rce** [reply]
I have solved this problem thx u >ㅅ<

```
helm upgrade --install agones agones/agones \\n--namespace agones-system \\n--set 'gameservers.namespaces={default,shulker-system}' \\n--set agones.allocator.install=true \\n--set agones.allocator.service.serviceType=ClusterIP
```

Thank you for your kind response

I was so fascinated by the idea of implementing Minecraft with Agones that I decided to back it.

Personally, I would love to do what you've created in EKS, one by one, but I've run into another problem.


~I ran the yaml file found here https://shulker.jeremylvln.fr/latest/guide/getting-started/your-first-cluster.html#a-spoon-of-minecraftcluster on my K8S and faced the error PreBind plugin “VolumeBinding”: binding volumes running on data-getting-started-redis-managed-0.~

I want to implement VolumeBinding with EFS, but I'm not sure how to do it... Or if that's not possible, what can I do with EBS?

The reason why we want to use EFS, which is easy to understand if you know anything about minifarms, is that we want to have one markserver per user to reflect their world information.

~[23:52:01 WARN] [io.shulkermc.proxyagent.velocity.ShulkerProxyAgentVelocity]: Fleet's external address was not found, the Service may not be ready yet or is not a LoadBalancer~

<img width="1196" alt="Screenshot 2024-08-30 at 9 00 38" src="https://github.com/user-attachments/assets/f484810b-d802-4e2a-a36d-22691329b910">

<img width="844" alt="Screenshot 2024-08-30 at 9 01 26" src="https://github.com/user-attachments/assets/835d7d87-2881-4e4c-aede-c21415cde4ce">

![Screenshot 2024-08-30 at 10 29 18](https://github.com/user-attachments/assets/af80cab1-ea0d-4e87-8350-8e5b4490fc2b)

<img width="430" alt="Screenshot 2024-08-30 at 10 30 12" src="https://github.com/user-attachments/assets/0b15adbf-283c-4bc2-8ed5-5615df20b105">

<img width="662" alt="Screenshot 2024-08-30 at 10 30 25" src="https://github.com/user-attachments/assets/e14f6090-6f2b-4b5e-b3a6-53f64f5546cf">

<img width="803" alt="Screenshot 2024-08-30 at 10 31 14" src="https://github.com/user-attachments/assets/efc1badf-3a5a-49fa-8e08-cdfaf4337e7e">

EKS Cluster is located in Public Subnet

__After lauching minecraftserver.yaml, I wasn't able to join my mc server. Just in case, I tried again, but I can't connect to the server.__


<img width="1032" alt="Screenshot 2024-08-30 at 12 20 13" src="https://github.com/user-attachments/assets/d6ceaa88-023a-48f8-b398-aeaafc750518">
<img width="622" alt="Screenshot 2024-08-30 at 12 21 45" src="https://github.com/user-attachments/assets/beac6198-9c50-4af4-8bf9-050b77b740f0">


kubectl describe pod -n shulker-tutorial public-fvfc2-7579z
<img width="1166" alt="Screenshot 2024-08-30 at 16 11 02" src="https://github.com/user-attachments/assets/801a6669-5cf1-45f4-820d-e34836ec155e">

I don't understand Agones or your project very well yet, so I have some additional questions.

I haved checked agones example (https://agones.dev/site/docs/examples/supertuxkart/) activated

When creating a minecraft server, is it possible to give the server and port number statically? And how does match making work in your project?

Sorry for asking so many questions at the beginning, but I'm very fascinated by your project.


If you are sensitive about what I write here, you can contact me on Discord (speedcore300 / Speedcore#3923).

**Reply by jeremylvln** [reply]
In order:

1. I'm not sure if you can mount a volume on EFS, only on block storage. If you want to store maps on EFS, I think you'll need to find a solution by yourself, Shulker does not do anything to the maps. It does, however, provide you a way to download a tar-gzipped archive containing a map (https://shulker.jeremylvln.fr/latest/guide/recipes/adding-custom-content.html)

2. You probably need to enable the Proxy Protocol if you can't connect. Because by default, the network ingress is re-routed from a node, thus hiding the real IP address. Because Velocity checks the IP you authenticated to Mojang & the IP the player is connecting from, if there is a mismatch, you'll be disconnected with this error message. I'll let you see how EKS manages the proxy protocol, here is how to enable it on your ProxyFleet: https://shulker.jeremylvln.fr/latest/guide/recipes/enabling-proxy-protocol.html

3. You can't connect directly to a server, you must use a proxy. That's why you can't manually assign IP & port

4. The match making is really not ready yet, it's something I work on time to time. But it's not ready

Thanks for your questions :)

**Reply by Rynf0rce** [reply]
Thank you for your response. 
I guess I shouldn't have expected it to be easy.
By the way, when I ran the Proxy.yaml on EKS, the pod restarted itself because there was no Readiness-rock file, and the proxy itself didn't run, what should I do about that?

kubectl describe pod -n shulker-tutorial public-fvfc2-7579z
<img width="1166" alt="Screenshot 2024-08-30 at 16 11 02" src="https://github.com/user-attachments/assets/801a6669-5cf1-45f4-820d-e34836ec155e">
